src/test/java/de/retest/recheck/persistence/bin/KryoPersistenceTest.java

```diff
@@ -0,0 +1,60 @@
+package de.retest.recheck.persistence.bin;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.mockito.Mockito;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.KryoException;
+
+import de.retest.recheck.persistence.IncompatibleReportVersionException;
+import de.retest.recheck.util.VersionProvider;
+
+class KryoPersistenceTest {
+
+	@Test
+	void roundtrip_should_work( @TempDir final Path temp ) throws IOException {
+		final Path file = temp.resolve( "test.test" );
+		Files.createFile( file );
+		final URI identifier = file.toUri();
+
+		final KryoPersistence<de.retest.recheck.test.Test> kryoPersistence = new KryoPersistence<>();
```

Comment:
Why not use a "real" `TestReport` instance like in the `KryoPersistenceTest` in review?

Reply:
Because it uses a lot of `builder` methods, that are not available in recheck (yet).